src/ageml/ui.py

```diff
@@ -816,29 +819,49 @@ def model_age(self, df, model, tag):
 
     def model_all(self):
         """Model age for each system and covariate on controls."""
-
+        # Get the test indices for the models
         # Iterate over covariate and system
         for covar in self.covars:
             for system in self.systems:
+                train_indices, test_indices = self.__get_test_indices(covar, system)
                 covar_tag = self.args.covar_name + "_" + str(covar) if self.flags["covarname"] else covar
                 tag = NameTag(covar=covar_tag, system=system)
                 # Generate model
                 ageml_model = self.generate_model()
-                self.models[covar][system], df_pred, self.betas[covar][system] = self.model_age(
-                    self.dfs["cn"][covar][system], ageml_model, tag=tag
-                )
+                data = self.dfs["cn"][covar][system]
+                ageml_model.train_indices = train_indices
+                ageml_model.test_indices = test_indices
+                model, df_pred, betas = self.model_age(data, ageml_model, tag=tag)
+                self.models[covar][system] = model
+                self.betas[covar][system] = betas
                 # Save predictions
                 df_pred = df_pred.drop(columns=["age"])
                 df_pred.rename(columns=lambda x: f"{x}_{system}", inplace=True)
                 self.preds["cn"][covar][system] = df_pred
 
-    def predict_age(
-        self,
-        df,
-        model,
-        tag: NameTag,
-        beta: np.ndarray = None,
-    ):
+    def __get_test_indices(self, covar, system):
+        """Gets and sets the train and test indices to keep track of them when running
+        the model_age pipeline. Necessary to inform the user that when hyperopt was
+        selected as the model, no CV was used, and instead a train test split was done.
+        """
+        data = self.dfs["cn"][covar][system]
+        X, y, _ = feature_extractor(data)
+
+        if self.args.model_type == "hyperopt":
+            # Split the data in training and test sets
+            # TODO: Substitute 0.2 factor by the user CLI argument
+            test_size = int(0.2 * y.shape[0])
+            indices = rng.permutation(X.shape[0])
+            train_indices = indices[:-test_size]
+            test_indices = indices[-test_size:]
+
+        else:
+            train_indices = list(range(X.shape[0]))
```

Comment:
why do we need this train and test indices? I see that they are the whole data structure.

Reply:
The reason behind it is to avoid writing `if` flow control statements every time I need to split the dataset depending on the model selection ('hyperopt' or not). It is an efficiency thing